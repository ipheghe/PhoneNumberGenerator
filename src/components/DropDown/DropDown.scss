@import '../../common/functions.scss';

.dropdown__container {
  position: relative;
  display: inline-block;

  &:hover {
    .dropdown__btn {
      background-color: #0d7d7c;
    }

    .dropdown__content {
      display: block;
    }
  }

  .dropdown__btn {
    display: flex;
    background-color: #57CCCb;
    padding: rem(16px);
    border: none;
    cursor: pointer;

    &:focus {
      outline: none;
    }
    
    p {
      color: white;
      font-size: rem(16px);
      padding:0;
      margin: 0;

      i {
        border: solid #FFF;
        border-width: 0 rem(3px) rem(3px) 0;
        display: inline-block;
        padding: rem(3px);
        margin-bottom: rem(2px);
        margin-right: rem(2px);
      }

      .down__arrow {
        transform: rotate(45deg);
        -webkit-transform: rotate(45deg);
      }
    }
  }

  .dropdown__content {
    display: none;
    position: absolute;
    background-color: #f1f1f1;
    min-width: rem(160px);
    box-shadow: 0 rem(8px) rem(16px) 0 rgba(0,0,0,0.2);
    z-index: 1;

    .dropdown__btn__item {
      color: black;
      padding: rem(12px) rem(16px);
      text-decoration: none;
      display: block;
      width: 100%;
      cursor: pointer;   

      &:hover {
        background-color: #ddd;
      }
    }
  } 
}

@media screen and (max-width: rem(450px)) {
  .dropdown__content {
    right: 0;
  }
}
