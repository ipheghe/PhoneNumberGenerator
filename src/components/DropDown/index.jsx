// react lib imports
import React from 'react';

// third-party-imports
import PropTypes from 'prop-types';

// styles
import './DropDown.scss';

/**
 * The renders the Drop Down component
 *
 * @returns {JSX} JSX
 */
const DropDown = ({ handleSortClick, className }) =>  (
  <div className={`dropdown__container ${className}`}>
    <button
      type="button"
      className="dropdown__btn"
    >
      <p>Sort By <span><i className="down__arrow"></i></span></p>
    </button>
    <div className="dropdown__content">
      <button
        type="button"
        className="dropdown__btn__item"
        onClick={(event) => handleSortClick(event, 'unique')}
      >
        Id
      </button>
      <button
        type="button"
        className="dropdown__btn__item"
        onClick={(event) => handleSortClick(event, 'highest')}
      >
        Highest Number
      </button>
      <button
        type="button"
        className="dropdown__btn__item"
        onClick={(event) => handleSortClick(event, 'lowest')}
      >
        Lowest Number
      </button>
    </div>
  </div>
);

DropDown.propTypes = {
  className: PropTypes.string,
  handleSortClick: PropTypes.func.isRequired,
};

DropDown.defaultProps = {
  className: '',
};

export default DropDown;
