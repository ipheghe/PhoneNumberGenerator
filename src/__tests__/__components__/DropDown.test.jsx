// react libraries
import React from 'react';

// third-party libraries
import { mount } from 'enzyme';
import render from 'react-test-renderer';
import { spy } from 'sinon';

// components
import DropDown from '../../components/DropDown';

const props = {
  handleSortClick: spy(),
  className: '',
};

describe('DropDown component', () => {
  it('should match component snapshot', () => {
    const tree = render.create(<DropDown {...props } />);
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const mountWrapper = mount(<DropDown {...props } />);
    expect(mountWrapper).toBeDefined();
    expect(mountWrapper.find('.dropdown__container').length).toBe(1);
    expect(mountWrapper.find('.dropdown__btn__item').length).toBe(3);
    expect(mountWrapper.find('.dropdown__btn__item').first().simulate('click'));
    expect(mountWrapper.exists()).toBe(true);
  });

  it('should call similate click event when any option is selected', () => {
    const shallowWrapper = mount(<DropDown {...props } />);
    const firstButtonClick = shallowWrapper.find('.dropdown__btn__item').at(1);
    firstButtonClick.simulate('click');
    const secondButtonClick = shallowWrapper.find('.dropdown__btn__item').at(2);
    secondButtonClick.simulate('click');
    expect(shallowWrapper.props().handleSortClick.called).toBe(true);
  });
});
