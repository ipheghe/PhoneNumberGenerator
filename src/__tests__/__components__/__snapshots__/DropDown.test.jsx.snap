// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown component should match component snapshot 1`] = `
<div
  className="dropdown__container "
>
  <button
    className="dropdown__btn"
    type="button"
  >
    <p>
      Sort By 
      <span>
        <i
          className="down__arrow"
        />
      </span>
    </p>
  </button>
  <div
    className="dropdown__content"
  >
    <button
      className="dropdown__btn__item"
      onClick={[Function]}
      type="button"
    >
      Id
    </button>
    <button
      className="dropdown__btn__item"
      onClick={[Function]}
      type="button"
    >
      Highest Number
    </button>
    <button
      className="dropdown__btn__item"
      onClick={[Function]}
      type="button"
    >
      Lowest Number
    </button>
  </div>
</div>
`;
